source/extensions/filters/network/http_connection_manager/dependency_manager.h

```diff
@@ -37,7 +35,7 @@ class DependencyManager {
   bool validDecodeDependencies();
 
 private:
-  std::vector<std::pair<absl::string_view,
+  std::vector<std::pair<const std::string&,
```

Comment:
The issue here is that you are relying on the lifetime of the registerFilter caller in internal code. It's probably fine, but you should either document that the arg must outlive the object or make this a `std::string` here (but it's fine to keep `const std::string&` in the param, it will create a copy which is desirable when lifetime is unclear).

Reply:
Ah, that works. So it takes a const reference to the function and makes a copy to store internally. Done.